### PR TITLE
ISSUE-176: Server group description mapping to enable clone/copy.

### PR DESCRIPTION
**Pull request by mtweten**
Also updates based on marathon-client model changes + various fixes:

issue-179 - Handle USER network types (setting the network name)

issue-189 - Don't send credentials back in the account, and add taskId in Instance to get around name clobbering

### CODE REVIEW

File: clouddriver-dcos/src/main/groovy/com/netflix/spinnaker/clouddriver/dcos/deploy/description/servergroup/DeployDcosServerGroupDescription.groovy [hunk 1]
```diff
@@ -73,19 +79,15 @@ class DeployDcosServerGroupDescription extends AbstractDcosCredentialsDescriptio
   @Canonical
   static class Docker {
     Image image
-    NetworkType network
+
+    // TODO this isn't actually used
+    String network
```
Comment (mtweten) [inline clouddriver-dcos/src/main/groovy/com/netflix/spinnaker/clouddriver/dcos/deploy/description/servergroup/DeployDcosServerGroupDescription.groovy:84]: I need to address these various TODOs around this network field. It exists in various places, but really in DC/OS only applies in the container/docker section. If there is no container/docker, then the network type is by default HOST in DC/OS. Currently everything works fine, it just seems like we don't necessarily need all this stuff.

File: clouddriver-dcos/src/main/groovy/com/netflix/spinnaker/clouddriver/dcos/deploy/util/DeployDcosServerGroupDescriptionToAppMapper.groovy [hunk 2]
```diff
@@ -85,17 +99,37 @@ class DeployDcosServerGroupDescriptionToAppMapper {
...
+              it
+            } : null
+
             taskKillGracePeriodSeconds = description.taskKillGracePeriodSeconds
             secrets = description.secrets
             requirePorts = description.requirePorts
             acceptedResourceRoles = description.acceptedResourceRoles
 
-            if (description.networkType != null && "BRIDGE" == description.networkType.type) {
+            // TODO I don't think this is correct
```
Comment (mtweten) [inline clouddriver-dcos/src/main/groovy/com/netflix/spinnaker/clouddriver/dcos/deploy/util/DeployDcosServerGroupDescriptionToAppMapper.groovy:131]: I think Trevin is working on this part.

File: clouddriver-dcos/src/main/groovy/com/netflix/spinnaker/clouddriver/dcos/deploy/util/AppToDeployDcosServerGroupDescriptionMapper.groovy [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+    app.container?.volumes?.each {
+      switch (it) {
+        case LocalVolume:
+          desc.dockerVolumes.add(new DeployDcosServerGroupDescription.DockerVolume(hostPath: ((LocalVolume) it).hostPath,
+            containerPath: it.containerPath, mode: it.mode))
+          break
+        case ExternalVolume:
+          def v = (ExternalVolume) it
+          desc.externalVolumes.add(new DeployDcosServerGroupDescription.ExternalVolume().with {
+            // TODO not really sure how to set "mode" and "options" on the ExternalStorage model (issue #200)
```
Comment (mtweten) [inline clouddriver-dcos/src/main/groovy/com/netflix/spinnaker/clouddriver/dcos/deploy/util/AppToDeployDcosServerGroupDescriptionMapper.groovy:179]: Looks like Trevin is looking into this.

File: clouddriver-dcos/src/main/groovy/com/netflix/spinnaker/clouddriver/dcos/deploy/description/servergroup/DeployDcosServerGroupDescription.groovy [hunk 4]
```diff
@@ -47,6 +52,7 @@ class DeployDcosServerGroupDescription extends AbstractDcosCredentialsDescriptio
     String type
   }
 
+  // TODO don't think we need a lot of this stuff (fromContext, fromTrigger, cluster, account)
```
Comment (mtweten) [inline clouddriver-dcos/src/main/groovy/com/netflix/spinnaker/clouddriver/dcos/deploy/description/servergroup/DeployDcosServerGroupDescription.groovy:55]: There was a lot of stuff existing on the image description that are never used: stageId, cluster, account, fromContext, fromTrigger. Is there any reason we still need them here?

**Comment by mtweten** [comment]
This is still showing the commits from the integration rebase snafu, not sure why since that has been corrected and this no longer accurately shows the actual state of the branches.

File: clouddriver-dcos/src/main/groovy/com/netflix/spinnaker/clouddriver/dcos/deploy/description/servergroup/DeployDcosServerGroupDescription.groovy (hunk 4 above)
Comment (TrevinTeacutter) [reply clouddriver-dcos/src/main/groovy/com/netflix/spinnaker/clouddriver/dcos/deploy/description/servergroup/DeployDcosServerGroupDescription.groovy:55]: There isn't any reason afaik, these are just used by Orca.

Comment (mtweten) [reply clouddriver-dcos/src/main/groovy/com/netflix/spinnaker/clouddriver/dcos/deploy/description/servergroup/DeployDcosServerGroupDescription.groovy:55]: Yeah, I think orca uses them in the input map that comes in from deck, but when it gets to clouddriver I think we can take them out of this model.

**Comment by mtweten** [comment]
Going to recreate this PR, since it seems all janked up.

**Pull request closed without merging**